crates/rattler_shell_helpers/src/lib.rs

```diff
@@ -0,0 +1,354 @@
+
+    let mut scripts = paths
+        .into_iter()
+        .filter(|r| r.is_ok())
+        .map(|r| r.unwrap().path())
+        .filter(|path| path.is_file() && path.extension() == Some(shell_type.suffix()))
+        .collect::<Vec<_>>();
+
+    scripts.sort();
+
+    Ok(scripts)
+}
+
+/// Collect all environment variables that are set in a conda environment.
+/// The environment variables are collected from the `state` file and the `env_vars.d` directory in the given prefix
+/// and are returned as a ordered map.
+///
+/// # Arguments
+///
+/// * `prefix` - The path to the root of the conda environment
+///
+/// # Returns
+///
+/// A map of environment variables
+///
+/// # Errors
+///
+/// If the `state` file or the `env_vars.d` directory cannot be read, an error is returned.
+fn collect_env_vars(prefix: &Path) -> anyhow::Result<IndexMap<String, String>> {
+    let state_file = prefix.join("conda-meta/state");
+    let pkg_env_var_dir = prefix.join("etc/conda/env_vars.d");
+    let mut env_vars = IndexMap::new();
+
+    if pkg_env_var_dir.exists() {
+        let env_var_files = pkg_env_var_dir.read_dir()?;
+        let mut env_var_files = env_var_files
+            .into_iter()
+            .filter(|r| r.is_ok())
+            // is safe because we filtered on `is_ok()`
+            .map(|r| r.unwrap().path())
+            .filter(|path| path.is_file())
+            .collect::<Vec<_>>();
+
+        env_var_files.sort();
+
+        // TODO figure out how to more safely `unwrap`?
+        let env_var_json: Vec<serde_json::Value> = env_var_files
+            .iter()
+            .map(|path| fs::read_to_string(path).unwrap())
+            .map(|json| serde_json::from_str(&json).unwrap())
+            .collect();
+
+        for env_var_json in env_var_json {
+            for (key, value) in env_var_json.as_object().unwrap() {
+                env_vars.insert(key.to_uppercase(), value.as_str().unwrap().to_string());
+            }
+        }
+    }
+
+    if state_file.exists() {
+        let state_json = fs::read_to_string(state_file).unwrap();
+
+        // load json but preserve the order of dicts - for this we use the serde preserve_order feature
+        let state_json: serde_json::Value = serde_json::from_str(&state_json).unwrap();
+        let state_env_vars = state_json["env_vars"].as_object().unwrap();
+
+        for (key, value) in state_env_vars {
+            // TODO log warning?
+            // println!("{}: {} (overwritten)", key, value.as_str().unwrap());
+            env_vars.insert(key.to_uppercase(), value.as_str().unwrap().to_string());
+        }
+    }
+    Ok(env_vars)
+}
+
+/// Return a vector of path entries that are prefixed with the given path.
+///
+/// # Arguments
+///
+/// * `prefix` - The path to prefix the path entries with
+/// * `operating_system` - The operating system that the path entries are for
+///
+/// # Returns
+///
+/// A vector of path entries
+fn prefix_path_entries(
```

Comment:
I dont quite get what this means, does this return where binaries can be stored?